bitcoin/src/address.rs

```diff
@@ -492,28 +518,30 @@ impl Payload {
         merkle_root: Option<TapNodeHash>,
     ) -> Payload {
         let (output_key, _parity) = internal_key.tap_tweak(secp, merkle_root);
-        Payload::WitnessProgram {
-            version: WitnessVersion::V1,
-            program: output_key.to_inner().serialize().to_vec(),
-        }
+        let prog = WitnessProgram::new(
+            WitnessVersion::V1,
+            output_key.to_inner().serialize().to_vec()
+        ).expect("taproot output key has len 32 <= 40");
+        Payload::WitnessProgram(prog)
     }
 
     /// Create a pay to taproot payload from a pre-tweaked output key.
     ///
     /// This method is not recommended for use and [Payload::p2tr()] should be used where possible.
     pub fn p2tr_tweaked(output_key: TweakedPublicKey) -> Payload {
-        Payload::WitnessProgram {
-            version: WitnessVersion::V1,
-            program: output_key.to_inner().serialize().to_vec(),
-        }
+        let prog = WitnessProgram::new(
+            WitnessVersion::V1,
+            output_key.to_inner().serialize().to_vec()
+        ).expect("taproot output key has len 32 <= 40");
+        Payload::WitnessProgram(prog)
     }
 
     /// Returns a byte slice of the payload
     pub fn as_bytes(&self) -> &[u8] {
         match self {
             Payload::ScriptHash(hash) => hash.as_ref(),
             Payload::PubkeyHash(hash) => hash.as_ref(),
-            Payload::WitnessProgram { program, .. } => program,
+            Payload::WitnessProgram(prog) => prog.program(),
```

Comment:
Unrelated: But this API is sketchy in how it is used. If anything I would expect the `witness_version` to get serialized in `as_bytes` method of the `PayLoad`. 

Should open an issue about it.

Reply:
Oh, I think I complained about this, IIRC this was introduced when implementing `is_related_to_pubkey`. The function shouldn't be `pub`.